framer/InputLayer.coffee

```diff
@@ -0,0 +1,148 @@
+Utils = require "./Utils"
+
+{Layer} = require "./Layer"
+{TextLayer} = require "./TextLayer"
+{Events} = require "./Events"
+
+Events.EnterKey  = "EnterKey"
+Events.BackspaceKey  = "BackspaceKey"
+Events.InputValueChange = "InputValueChange"
+Events.InputFocus = "InputFocus"
+Events.InputBlur = "InputBlur"
+
+class exports.InputLayer extends TextLayer
+
+	constructor: (options={}) ->
+
+		_.defaults options,
+			backgroundColor: "#FFF"
+			width: 500
+			height: 100
+
+		super options
+
+		# Layer containing input element
+		@input = new Layer
+			backgroundColor: "transparent"
+			x: 30
+			width: @width - 60
+			height: @height
+			parent: @
+			name: "input"
+
+		if @multiLine
+			@_inputElement = document.createElement("textarea")
+			@input.y = 25
+		else
+			@_inputElement = document.createElement("input")
+
+		# The id serves to differentiate multiple input elements from one another.
+		# To allow styling the placeholder colors of seperate elements.
+		@_id = Utils.round(Utils.randomNumber(0, 100))
+		@_inputElement.className = "input" + @_id
```

Comment:
This is random, but not random enough. You can either:
- Make the random id really unique by making it a [UUID](https://en.wikipedia.org/wiki/Universally_unique_identifier)
- Keep track of the number of InputLayers globally, so you can increment the value and have it always be unique (until you run out of integers)

Or just look at the definition of the `@id` property on `BaseClass` and realise that every instance of a class that subclasses from it already has a unique id that you can use for this purpose.